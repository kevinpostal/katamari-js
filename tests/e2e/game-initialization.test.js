/**
 * Snapshot tests for game initialization state
 * Tests game initialization state consistency and configuration snapshots
 */

import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { createMockGameState, setupGameEnvironment, validateGameState } from '../helpers/game-helpers.js';

describe('Game Initialization Snapshots', () => {
    let gameEnvironment;

    beforeEach(() => {
        // Create a clean game environment for each test
        gameEnvironment = setupGameEnvironment({
            includePhysics: true,
            includeRendering: true,
            includeAudio: true,
            itemCount: 0, // Start with empty world for initialization tests
            initialKatamariSize: 1.0
        });
    });

    afterEach(() => {
        if (gameEnvironment?.dispose) {
            gameEnvironment.dispose();
        }
    });

    describe('Initial Game State Snapshots', () => {
        it('should match initial game state snapshot', () => {
            const initialState = createMockGameState('initial');
            
            // Validate state consistency before snapshot
            const validation = validateGameState(initialState);
            expect(validation.valid).toBe(true);
            
            // Snapshot the initial state structure
            expect(initialState).toMatchSnapshot('initial-game-state');
        });

        it('should match initial scene configuration snapshot', () => {
            const sceneConfig = {
                initialized: false,
                objectCount: 0,
                lightingSetup: false,
                background: null,
                fog: null,
                children: [],
                lighting: {
                    ambient: {
                        color: 0x404040,
                        intensity: 0.4
                    },
                    directional: {
                        color: 0xffffff,
                        intensity: 1.0,
                        position: { x: 10, y: 10, z: 5 },
                        castShadow: true
                    },
                    hemisphere: {
                        skyColor: 0x87CEEB,
                        groundColor: 0x362d1d,
                        intensity: 0.6
                    }
                }
            };

            expect(sceneConfig).toMatchSnapshot('initial-scene-configuration');
        });

        it('should match initial physics world configuration snapshot', () => {
            const physicsConfig = {
                worldCreated: false,
                gravity: { x: 0, y: -9.82, z: 0 },
                broadphase: 'NaiveBroadphase',
                solver: {
                    iterations: 10,
                    tolerance: 0.0001
                },
                bodies: [],
                bodyCount: 0,
                activeCollisions: 0,
                contactMaterial: {
                    friction: 0.3,
                    restitution: 0.3,
                    contactEquationStiffness: 1e8,
                    contactEquationRelaxation: 3
                }
            };

            expect(physicsConfig).toMatchSnapshot('initial-physics-configuration');
        });

        it('should match initial katamari configuration snapshot', () => {
            const katamariConfig = {
                position: { x: 0, y: 0, z: 0 },
                size: 1.0,
                velocity: { x: 0, y: 0, z: 0 },
                collectedItems: 0,
                mass: 1.0,
                material: {
                    friction: 0.4,
                    restitution: 0.3
                },
                geometry: {
                    type: 'SphereGeometry',
                    radius: 1.0,
                    widthSegments: 32,
                    heightSegments: 16
                },
                visual: {
                    color: 0x00ff00,
                    metalness: 0.1,
                    roughness: 0.7,
                    castShadow: true,
                    receiveShadow: true
                }
            };

            expect(katamariConfig).toMatchSnapshot('initial-katamari-configuration');
        });

        it('should match initial audio system configuration snapshot', () => {
            const audioConfig = {
                context: {
                    state: 'suspended',
                    sampleRate: 44100,
                    destination: 'speakers'
                },
                synthesizers: {
                    membrane: {
                        type: 'MembraneSynth',
                        volume: -12,
                        envelope: {
                            attack: 0.001,
                            decay: 0.4,
                            sustain: 0.01,
                            release: 1.4
                        },
                        octaves: 10,
                        oscillator: {
                            type: 'sine'
                        }
                    },
                    noise: {
                        type: 'NoiseSynth',
                        volume: -20,
                        envelope: {
                            attack: 0.005,
                            decay: 0.1,
                            sustain: 0.0
                        },
                        noise: {
                            type: 'white'
                        }
                    },
                    basic: {
                        type: 'Synth',
                        volume: -15,
                        envelope: {
                            attack: 0.005,
                            decay: 0.1,
                            sustain: 0.3,
                            release: 1
                        },
                        oscillator: {
                            type: 'triangle'
                        }
                    }
                },
                effects: {
                    reverb: {
                        roomSize: 0.7,
                        dampening: 3000
                    },
                    filter: {
                        frequency: 350,
                        type: 'lowpass'
                    }
                }
            };

            expect(audioConfig).toMatchSnapshot('initial-audio-configuration');
        });
    });

    describe('Default Game Settings Snapshots', () => {
        it('should match default game constants snapshot', () => {
            const gameConstants = {
                PHYSICS: {
                    GRAVITY: -9.82,
                    TIMESTEP: 1/60,
                    MAX_SUBSTEPS: 3,
                    KATAMARI_FRICTION: 0.4,
                    KATAMARI_RESTITUTION: 0.3,
                    ITEM_FRICTION: 0.3,
                    ITEM_RESTITUTION: 0.3
                },
                RENDERING: {
                    FOV: 75,
                    NEAR_PLANE: 0.1,
                    FAR_PLANE: 1000,
                    SHADOW_MAP_SIZE: 2048,
                    MAX_INSTANCES: 1000,
                    FRUSTUM_CULLING: true
                },
                GAMEPLAY: {
                    INITIAL_KATAMARI_SIZE: 1.0,
                    SIZE_GROWTH_FACTOR: 0.1,
                    COLLECTION_THRESHOLD: 0.8,
                    LEVEL_PROGRESSION_MULTIPLIER: 2.0,
                    MAX_ITEMS_PER_LEVEL: 500
                },
                CONTROLS: {
                    KEYBOARD_SENSITIVITY: 1.0,
                    TOUCH_SENSITIVITY: 0.8,
                    GYRO_SENSITIVITY: 0.5,
                    MOVEMENT_DAMPING: 0.9
                },
                AUDIO: {
                    MASTER_VOLUME: 0.7,
                    SFX_VOLUME: 0.8,
                    AMBIENT_VOLUME: 0.5,
                    COLLECTION_SOUND_VARIATION: 0.3
                }
            };

            expect(gameConstants).toMatchSnapshot('default-game-constants');
        });

        it('should match default level configuration snapshot', () => {
            const levelConfig = {
                currentLevel: 1,
                theme: 'earth',
                targetSize: 10.0,
                itemsGenerated: 0,
                themes: {
                    earth: {
                        name: 'Earth',
                        backgroundColor: 0x87CEEB,
                        fogColor: 0x87CEEB,
                        fogNear: 50,
                        fogFar: 200,
                        items: ['leaf', 'stick', 'rock', 'flower', 'mushroom'],
                        itemSizes: [0.1, 0.2, 0.3, 0.15, 0.25],
                        itemColors: [0x228B22, 0x8B4513, 0x696969, 0xFF69B4, 0xDEB887],
                        spawnRadius: 50,
                        itemDensity: 0.8
                    },
                    urban: {
                        name: 'Urban',
                        backgroundColor: 0x708090,
                        fogColor: 0x708090,
                        fogNear: 30,
                        fogFar: 150,
                        items: ['coin', 'paperclip', 'pen', 'book', 'chair'],
                        itemSizes: [0.05, 0.08, 0.12, 0.4, 0.8],
                        itemColors: [0xFFD700, 0xC0C0C0, 0x000080, 0x8B4513, 0x654321],
                        spawnRadius: 40,
                        itemDensity: 1.0
                    },
                    space: {
                        name: 'Space',
                        backgroundColor: 0x000011,
                        fogColor: 0x000011,
                        fogNear: 100,
                        fogFar: 500,
                        items: ['asteroid', 'satellite', 'planet', 'star', 'galaxy'],
                        itemSizes: [0.5, 1.0, 5.0, 0.3, 20.0],
                        itemColors: [0x696969, 0xC0C0C0, 0x4169E1, 0xFFFF00, 0x9370DB],
                        spawnRadius: 100,
                        itemDensity: 0.5
                    }
                },
                progression: {
                    sizeMultipliers: [1, 2.5, 6.25, 15.625, 39.0625],
                    levelThresholds: [10, 25, 62.5, 156.25, 390.625],
                    itemCountMultipliers: [1, 1.5, 2.25, 3.375, 5.0625]
                }
            };

            expect(levelConfig).toMatchSnapshot('default-level-configuration');
        });

        it('should match default input configuration snapshot', () => {
            const inputConfig = {
                keyboard: {
                    enabled: true,
                    keys: {
                        forward: ['KeyW', 'ArrowUp'],
                        backward: ['KeyS', 'ArrowDown'],
                        left: ['KeyA', 'ArrowLeft'],
                        right: ['KeyD', 'ArrowRight'],
                        debug: ['KeyF1'],
                        pause: ['Space', 'Escape']
                    },
                    sensitivity: 1.0,
                    deadzone: 0.1
                },
                touch: {
                    enabled: true,
                    sensitivity: 0.8,
                    deadzone: 0.15,
                    maxDistance: 100,
                    visualFeedback: true
                },
                gyroscope: {
                    enabled: false,
                    sensitivity: 0.5,
                    deadzone: 0.2,
                    calibration: { x: 0, y: 0, z: 0 },
                    smoothing: 0.8
                },
                gamepad: {
                    enabled: true,
                    deadzone: 0.2,
                    sensitivity: 1.0,
                    buttonMapping: {
                        pause: 9, // Start button
                        debug: 8  // Select button
                    }
                }
            };

            expect(inputConfig).toMatchSnapshot('default-input-configuration');
        });
    });

    describe('Game Environment Initialization Snapshots', () => {
        it('should match complete game environment initialization snapshot', () => {
            const environmentSnapshot = {
                scene: {
                    type: 'Scene',
                    children: [],
                    background: null,
                    fog: null,
                    overrideMaterial: null,
                    autoUpdate: true
                },
                camera: {
                    type: 'PerspectiveCamera',
                    fov: 75,
                    aspect: window.innerWidth / window.innerHeight || 1,
                    near: 0.1,
                    far: 1000,
                    position: { x: 0, y: 5, z: 10 },
                    target: { x: 0, y: 0, z: 0 }
                },
                renderer: {
                    type: 'WebGLRenderer',
                    antialias: true,
                    shadowMap: {
                        enabled: true,
                        type: 'PCFSoftShadowMap'
                    },
                    outputEncoding: 'sRGBEncoding',
                    toneMapping: 'ACESFilmicToneMapping',
                    toneMappingExposure: 1.0
                },
                physics: {
                    world: {
                        gravity: { x: 0, y: -9.82, z: 0 },
                        broadphase: 'NaiveBroadphase',
                        solver: {
                            iterations: 10,
                            tolerance: 0.0001
                        }
                    },
                    materials: {
                        katamari: {
                            friction: 0.4,
                            restitution: 0.3
                        },
                        item: {
                            friction: 0.3,
                            restitution: 0.3
                        },
                        ground: {
                            friction: 0.8,
                            restitution: 0.1
                        }
                    }
                },
                audio: {
                    initialized: false,
                    context: null,
                    synthesizers: {},
                    effects: {}
                },
                gameState: gameEnvironment.gameState
            };

            expect(environmentSnapshot).toMatchSnapshot('complete-game-environment-initialization');
        });

        it('should match performance monitoring initialization snapshot', () => {
            const performanceConfig = {
                monitoring: {
                    enabled: true,
                    sampleRate: 60, // samples per second
                    historyLength: 300, // 5 seconds at 60fps
                    thresholds: {
                        frameRate: {
                            minimum: 55,
                            target: 60,
                            warning: 50,
                            critical: 30
                        },
                        memory: {
                            heapWarning: 100 * 1024 * 1024, // 100MB
                            heapCritical: 200 * 1024 * 1024, // 200MB
                            leakDetection: true
                        },
                        physics: {
                            stepTimeWarning: 5.0, // milliseconds
                            stepTimeCritical: 10.0
                        },
                        rendering: {
                            drawCallWarning: 1000,
                            drawCallCritical: 2000,
                            triangleWarning: 100000,
                            triangleCritical: 500000
                        }
                    }
                },
                metrics: {
                    frameRate: {
                        current: 0,
                        average: 0,
                        minimum: Infinity,
                        maximum: 0,
                        history: []
                    },
                    memory: {
                        heapUsed: 0,
                        heapTotal: 0,
                        external: 0,
                        arrayBuffers: 0,
                        history: []
                    },
                    timing: {
                        physicsStep: 0,
                        renderTime: 0,
                        updateTime: 0,
                        totalFrameTime: 0,
                        history: []
                    }
                }
            };

            expect(performanceConfig).toMatchSnapshot('performance-monitoring-initialization');
        });
    });
});