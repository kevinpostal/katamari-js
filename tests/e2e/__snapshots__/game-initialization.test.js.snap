// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Initialization Snapshots > Default Game Settings Snapshots > should match default game constants snapshot > default-game-constants 1`] = `
{
  "AUDIO": {
    "AMBIENT_VOLUME": 0.5,
    "COLLECTION_SOUND_VARIATION": 0.3,
    "MASTER_VOLUME": 0.7,
    "SFX_VOLUME": 0.8,
  },
  "CONTROLS": {
    "GYRO_SENSITIVITY": 0.5,
    "KEYBOARD_SENSITIVITY": 1,
    "MOVEMENT_DAMPING": 0.9,
    "TOUCH_SENSITIVITY": 0.8,
  },
  "GAMEPLAY": {
    "COLLECTION_THRESHOLD": 0.8,
    "INITIAL_KATAMARI_SIZE": 1,
    "LEVEL_PROGRESSION_MULTIPLIER": 2,
    "MAX_ITEMS_PER_LEVEL": 500,
    "SIZE_GROWTH_FACTOR": 0.1,
  },
  "PHYSICS": {
    "GRAVITY": -9.82,
    "ITEM_FRICTION": 0.3,
    "ITEM_RESTITUTION": 0.3,
    "KATAMARI_FRICTION": 0.4,
    "KATAMARI_RESTITUTION": 0.3,
    "MAX_SUBSTEPS": 3,
    "TIMESTEP": 0.016666666666666666,
  },
  "RENDERING": {
    "FAR_PLANE": 1000,
    "FOV": 75,
    "FRUSTUM_CULLING": true,
    "MAX_INSTANCES": 1000,
    "NEAR_PLANE": 0.1,
    "SHADOW_MAP_SIZE": 2048,
  },
}
`;

exports[`Game Initialization Snapshots > Default Game Settings Snapshots > should match default input configuration snapshot > default-input-configuration 1`] = `
{
  "gamepad": {
    "buttonMapping": {
      "debug": 8,
      "pause": 9,
    },
    "deadzone": 0.2,
    "enabled": true,
    "sensitivity": 1,
  },
  "gyroscope": {
    "calibration": {
      "x": 0,
      "y": 0,
      "z": 0,
    },
    "deadzone": 0.2,
    "enabled": false,
    "sensitivity": 0.5,
    "smoothing": 0.8,
  },
  "keyboard": {
    "deadzone": 0.1,
    "enabled": true,
    "keys": {
      "backward": [
        "KeyS",
        "ArrowDown",
      ],
      "debug": [
        "KeyF1",
      ],
      "forward": [
        "KeyW",
        "ArrowUp",
      ],
      "left": [
        "KeyA",
        "ArrowLeft",
      ],
      "pause": [
        "Space",
        "Escape",
      ],
      "right": [
        "KeyD",
        "ArrowRight",
      ],
    },
    "sensitivity": 1,
  },
  "touch": {
    "deadzone": 0.15,
    "enabled": true,
    "maxDistance": 100,
    "sensitivity": 0.8,
    "visualFeedback": true,
  },
}
`;

exports[`Game Initialization Snapshots > Default Game Settings Snapshots > should match default level configuration snapshot > default-level-configuration 1`] = `
{
  "currentLevel": 1,
  "itemsGenerated": 0,
  "progression": {
    "itemCountMultipliers": [
      1,
      1.5,
      2.25,
      3.375,
      5.0625,
    ],
    "levelThresholds": [
      10,
      25,
      62.5,
      156.25,
      390.625,
    ],
    "sizeMultipliers": [
      1,
      2.5,
      6.25,
      15.625,
      39.0625,
    ],
  },
  "targetSize": 10,
  "theme": "earth",
  "themes": {
    "earth": {
      "backgroundColor": 8900331,
      "fogColor": 8900331,
      "fogFar": 200,
      "fogNear": 50,
      "itemColors": [
        2263842,
        9127187,
        6908265,
        16738740,
        14596231,
      ],
      "itemDensity": 0.8,
      "itemSizes": [
        0.1,
        0.2,
        0.3,
        0.15,
        0.25,
      ],
      "items": [
        "leaf",
        "stick",
        "rock",
        "flower",
        "mushroom",
      ],
      "name": "Earth",
      "spawnRadius": 50,
    },
    "space": {
      "backgroundColor": 17,
      "fogColor": 17,
      "fogFar": 500,
      "fogNear": 100,
      "itemColors": [
        6908265,
        12632256,
        4286945,
        16776960,
        9662683,
      ],
      "itemDensity": 0.5,
      "itemSizes": [
        0.5,
        1,
        5,
        0.3,
        20,
      ],
      "items": [
        "asteroid",
        "satellite",
        "planet",
        "star",
        "galaxy",
      ],
      "name": "Space",
      "spawnRadius": 100,
    },
    "urban": {
      "backgroundColor": 7372944,
      "fogColor": 7372944,
      "fogFar": 150,
      "fogNear": 30,
      "itemColors": [
        16766720,
        12632256,
        128,
        9127187,
        6636321,
      ],
      "itemDensity": 1,
      "itemSizes": [
        0.05,
        0.08,
        0.12,
        0.4,
        0.8,
      ],
      "items": [
        "coin",
        "paperclip",
        "pen",
        "book",
        "chair",
      ],
      "name": "Urban",
      "spawnRadius": 40,
    },
  },
}
`;

exports[`Game Initialization Snapshots > Game Environment Initialization Snapshots > should match complete game environment initialization snapshot > complete-game-environment-initialization 1`] = `
{
  "audio": {
    "context": null,
    "effects": {},
    "initialized": false,
    "synthesizers": {},
  },
  "camera": {
    "aspect": 1.3333333333333333,
    "far": 1000,
    "fov": 75,
    "near": 0.1,
    "position": {
      "x": 0,
      "y": 5,
      "z": 10,
    },
    "target": {
      "x": 0,
      "y": 0,
      "z": 0,
    },
    "type": "PerspectiveCamera",
  },
  "gameState": {
    "katamari": {
      "collectedItems": 0,
      "position": {
        "x": 0,
        "y": 0,
        "z": 0,
      },
      "size": 1,
      "velocity": {
        "x": 0,
        "y": 0,
        "z": 0,
      },
    },
    "level": {
      "currentLevel": 1,
      "itemsGenerated": 0,
      "targetSize": 10,
      "theme": "earth",
    },
    "physics": {
      "activeCollisions": 0,
      "bodyCount": 0,
      "worldCreated": false,
    },
    "scene": {
      "initialized": false,
      "lightingSetup": false,
      "objectCount": 0,
    },
  },
  "physics": {
    "materials": {
      "ground": {
        "friction": 0.8,
        "restitution": 0.1,
      },
      "item": {
        "friction": 0.3,
        "restitution": 0.3,
      },
      "katamari": {
        "friction": 0.4,
        "restitution": 0.3,
      },
    },
    "world": {
      "broadphase": "NaiveBroadphase",
      "gravity": {
        "x": 0,
        "y": -9.82,
        "z": 0,
      },
      "solver": {
        "iterations": 10,
        "tolerance": 0.0001,
      },
    },
  },
  "renderer": {
    "antialias": true,
    "outputEncoding": "sRGBEncoding",
    "shadowMap": {
      "enabled": true,
      "type": "PCFSoftShadowMap",
    },
    "toneMapping": "ACESFilmicToneMapping",
    "toneMappingExposure": 1,
    "type": "WebGLRenderer",
  },
  "scene": {
    "autoUpdate": true,
    "background": null,
    "children": [],
    "fog": null,
    "overrideMaterial": null,
    "type": "Scene",
  },
}
`;

exports[`Game Initialization Snapshots > Game Environment Initialization Snapshots > should match performance monitoring initialization snapshot > performance-monitoring-initialization 1`] = `
{
  "metrics": {
    "frameRate": {
      "average": 0,
      "current": 0,
      "history": [],
      "maximum": 0,
      "minimum": Infinity,
    },
    "memory": {
      "arrayBuffers": 0,
      "external": 0,
      "heapTotal": 0,
      "heapUsed": 0,
      "history": [],
    },
    "timing": {
      "history": [],
      "physicsStep": 0,
      "renderTime": 0,
      "totalFrameTime": 0,
      "updateTime": 0,
    },
  },
  "monitoring": {
    "enabled": true,
    "historyLength": 300,
    "sampleRate": 60,
    "thresholds": {
      "frameRate": {
        "critical": 30,
        "minimum": 55,
        "target": 60,
        "warning": 50,
      },
      "memory": {
        "heapCritical": 209715200,
        "heapWarning": 104857600,
        "leakDetection": true,
      },
      "physics": {
        "stepTimeCritical": 10,
        "stepTimeWarning": 5,
      },
      "rendering": {
        "drawCallCritical": 2000,
        "drawCallWarning": 1000,
        "triangleCritical": 500000,
        "triangleWarning": 100000,
      },
    },
  },
}
`;

exports[`Game Initialization Snapshots > Initial Game State Snapshots > should match initial audio system configuration snapshot > initial-audio-configuration 1`] = `
{
  "context": {
    "destination": "speakers",
    "sampleRate": 44100,
    "state": "suspended",
  },
  "effects": {
    "filter": {
      "frequency": 350,
      "type": "lowpass",
    },
    "reverb": {
      "dampening": 3000,
      "roomSize": 0.7,
    },
  },
  "synthesizers": {
    "basic": {
      "envelope": {
        "attack": 0.005,
        "decay": 0.1,
        "release": 1,
        "sustain": 0.3,
      },
      "oscillator": {
        "type": "triangle",
      },
      "type": "Synth",
      "volume": -15,
    },
    "membrane": {
      "envelope": {
        "attack": 0.001,
        "decay": 0.4,
        "release": 1.4,
        "sustain": 0.01,
      },
      "octaves": 10,
      "oscillator": {
        "type": "sine",
      },
      "type": "MembraneSynth",
      "volume": -12,
    },
    "noise": {
      "envelope": {
        "attack": 0.005,
        "decay": 0.1,
        "sustain": 0,
      },
      "noise": {
        "type": "white",
      },
      "type": "NoiseSynth",
      "volume": -20,
    },
  },
}
`;

exports[`Game Initialization Snapshots > Initial Game State Snapshots > should match initial game state snapshot > initial-game-state 1`] = `
{
  "katamari": {
    "collectedItems": 0,
    "position": {
      "x": 0,
      "y": 0,
      "z": 0,
    },
    "size": 1,
    "velocity": {
      "x": 0,
      "y": 0,
      "z": 0,
    },
  },
  "level": {
    "currentLevel": 1,
    "itemsGenerated": 0,
    "targetSize": 10,
    "theme": "earth",
  },
  "physics": {
    "activeCollisions": 0,
    "bodyCount": 0,
    "worldCreated": false,
  },
  "scene": {
    "initialized": false,
    "lightingSetup": false,
    "objectCount": 0,
  },
}
`;

exports[`Game Initialization Snapshots > Initial Game State Snapshots > should match initial katamari configuration snapshot > initial-katamari-configuration 1`] = `
{
  "collectedItems": 0,
  "geometry": {
    "heightSegments": 16,
    "radius": 1,
    "type": "SphereGeometry",
    "widthSegments": 32,
  },
  "mass": 1,
  "material": {
    "friction": 0.4,
    "restitution": 0.3,
  },
  "position": {
    "x": 0,
    "y": 0,
    "z": 0,
  },
  "size": 1,
  "velocity": {
    "x": 0,
    "y": 0,
    "z": 0,
  },
  "visual": {
    "castShadow": true,
    "color": 65280,
    "metalness": 0.1,
    "receiveShadow": true,
    "roughness": 0.7,
  },
}
`;

exports[`Game Initialization Snapshots > Initial Game State Snapshots > should match initial physics world configuration snapshot > initial-physics-configuration 1`] = `
{
  "activeCollisions": 0,
  "bodies": [],
  "bodyCount": 0,
  "broadphase": "NaiveBroadphase",
  "contactMaterial": {
    "contactEquationRelaxation": 3,
    "contactEquationStiffness": 100000000,
    "friction": 0.3,
    "restitution": 0.3,
  },
  "gravity": {
    "x": 0,
    "y": -9.82,
    "z": 0,
  },
  "solver": {
    "iterations": 10,
    "tolerance": 0.0001,
  },
  "worldCreated": false,
}
`;

exports[`Game Initialization Snapshots > Initial Game State Snapshots > should match initial scene configuration snapshot > initial-scene-configuration 1`] = `
{
  "background": null,
  "children": [],
  "fog": null,
  "initialized": false,
  "lighting": {
    "ambient": {
      "color": 4210752,
      "intensity": 0.4,
    },
    "directional": {
      "castShadow": true,
      "color": 16777215,
      "intensity": 1,
      "position": {
        "x": 10,
        "y": 10,
        "z": 5,
      },
    },
    "hemisphere": {
      "groundColor": 3550493,
      "intensity": 0.6,
      "skyColor": 8900331,
    },
  },
  "lightingSetup": false,
  "objectCount": 0,
}
`;
